### DISCUSSION TITLE
Locking and unlocking a body part during simulation

### DISCUSSION THREAD

**Discussion by alvarobelmontebaeza** [isaac-sim/IsaacLab#430] [Q&A]
Hi! I want to lock a certain part of my robot's body at a certain point during simulation. For example, imagine a manipulator reaches a grasping point, and then gets fixed to a certain structure. I want to then lock the position of the end-effector so that the robot joints can still move, but the end-effector will remain fixed at its position. 

How could I achieve this behavior? Thanks in advance for your answers!

**Comment by abizovnuralem** [comment]
Are you planning to lock it from the beginning of training or during?

**Reply by alvarobelmontebaeza** [reply]
During training. I want to lock it and unlock it when certain conditions are met. 

I know that if it was only at the start of training I could set the body as a Kinematic body, but haven't found a way to do so dynamically during training.
